file-source/src/main/java/org/springframework/cloud/stream/module/file/source/FileSource.java

```diff
@@ -0,0 +1,128 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.module.file.source;
+
+import java.io.File;
+import java.util.Collections;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.config.SpelExpressionConverterConfiguration;
+import org.springframework.cloud.stream.messaging.Source;
+import org.springframework.cloud.stream.module.MaxMessagesProperties;
+import org.springframework.cloud.stream.module.PeriodicTriggerConfiguration;
+import org.springframework.cloud.stream.module.file.FileConsumerProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+import org.springframework.integration.dsl.IntegrationFlow;
+import org.springframework.integration.dsl.IntegrationFlowBuilder;
+import org.springframework.integration.dsl.IntegrationFlows;
+import org.springframework.integration.dsl.SourcePollingChannelAdapterSpec;
+import org.springframework.integration.dsl.core.Pollers;
+import org.springframework.integration.dsl.file.FileInboundChannelAdapterSpec;
+import org.springframework.integration.dsl.file.Files;
+import org.springframework.integration.dsl.support.Consumer;
+import org.springframework.integration.file.FileReadingMessageSource;
+import org.springframework.integration.file.splitter.FileSplitter;
+import org.springframework.integration.file.transformer.FileToByteArrayTransformer;
+import org.springframework.integration.scheduling.PollerMetadata;
+import org.springframework.messaging.MessageHeaders;
+import org.springframework.scheduling.Trigger;
+import org.springframework.util.StringUtils;
+
+/**
+ * Creates a {@link FileReadingMessageSource} bean and registers it as a
+ * Inbound Channel Adapter that sends messages from the Source output channel.
+ *
+ * @author Gary Russell
+ */
+@EnableBinding(Source.class)
+@Import({PeriodicTriggerConfiguration.class, SpelExpressionConverterConfiguration.class})
+@EnableConfigurationProperties({ FileSourceProperties.class,
+	FileConsumerProperties.class, MaxMessagesProperties.class})
+public class FileSource {
+
+	@Autowired
+	private FileSourceProperties properties;
+
+	@Autowired
+	private FileConsumerProperties fileConsumerProperties;
+
+	@Autowired
+	MaxMessagesProperties maxMessagesProperties;
+
+	@Autowired
+	Trigger trigger;
+
+	@Bean
+	public PollerMetadata poller() {
+		return Pollers.trigger(trigger).maxMessagesPerPoll(this.maxMessagesProperties.getMaxMessages()).get();
+	}
+
+	@Autowired
+	Source source;
```

Comment:
Can some one explain me why do we need such a simple interface injection, if it doesn't have so much power only `output()`? 
Why just don't inject that `MessageChannel output` ?
Thanks